### PR TITLE
Added public API to add third-party adapters

### PR DESCRIPTION
**Pull request by davej**
Re: #177 #176
 
Usage:
```js
// accord.addAdapter(adapterName, adapterPath, [nodeModuleName])
// If `nodeModuleName` is not specified then it is assumed to be the same as `adapterName`
accord.addAdapter('babel-legacy', require.resolve('accord-babel-legacy'), 'babel');

// Then use the adapter as you would with any other adapter...
accord.load('babel-legacy');
```

This is a first pass, happy to make any changes if you would like it to work a bit differently. :)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,9 +22,9 @@
   },
   "devDependencies": {
     "LiveScript": "latest",
-    "typescript-compiler": "^1.4",
+    "accord-babel-legacy": "^1.0.0",
     "acorn": "latest",
-    "babel": "latest",
+    "babel": "^5.8.38",
```
Comment (jenius) [inline package.json]: So this is not going to allow us to test against current babel, which is a larger issue. In order to get both running I might run this through [polytest](https://github.com/carrot/polytest), which we already have wired up in this library...

Comment (davej) [inline package.json]: Are you sure about this? Works fine for me and the tests pass on my machine. The normal Babel adapter seems to only use `babel-core` and the legacy babel adapter uses the `babel` module.

Comment (jenius) [inline package.json]: Ah that's right name change comes in to your advantage here haha. Never mind!

**Comment by davej** [comment]
So tests passing on Travis now.

By the way, it would be just as easy to include to include babel@5 support in accord under a different name (e.g.. `babel-legacy`). Whatever you decide, I still think this kind of API would be useful though because it makes it easy to hack on adapters without needing to fork Accord.

**Comment by jenius** [comment]
Looking good. Clean up the commits a little, then we can merge?

**Comment by davej** [comment]
Done.

**Comment by jenius** [comment]
Hey @davej -- unfortunately there is an issue in this code that is breaking a large number of our projects. The line that was removed and commented above is necessary in order to correctly map between adapters. At the same time, it appears as if removing the line maps `babel-legacy` over `babel` and causes the tests to fail.

I'm not exactly sure how to resolve this issue, so for now I need to roll back the change. If you have any ideas on how to get around this, would be happy to chat about them and/or look at a PR with a patch!

**Comment by davej** [comment]
Ok, I'll take a look tomorrow.

Yeah the PR was initially just a draft so we could find a good API... I lost sight of this a bit and should have asked that we flesh it out a bit more before merge.

**Pull request merged**